var.mk

```diff
@@ -14,14 +14,29 @@ PROGRAMATICA_SRC_FILE ?= \
 # The local file gets tried first, and if not usable the remote on gets fetched.
 # If both are unset or set to an empty string, programatica support is skipped.
 
-GHCVERSION := $(shell ghc --numeric-version)
+OSNAME := $(shell uname -s)
+OSVERS := $(shell uname -v 2>/dev/null)
+
+STACK := stack
+STACK_EXEC := $(STACK) exec --
+
+ifeq "$(OSNAME)" "Darwin"
+    STACK_DEPENDENCIES_FLAGS := --flag gtk:have-quartz-gtk
+else
+    STACK_DEPENDENCIES_FLAGS :=
+endif
+
+# Dummy variables to install GHC and install the Haskell-dependencies.  These
+# must be called before ghc or ghc-pkg is called because the packages need to
+# be installed before they are checked.
+STACK_SETUP := $(shell $(STACK) setup > /dev/null)
+STACK_INSTALL_DEPENDENCIES := $(shell $(STACK) build --only-dependencies $(STACK_DEPENDENCIES_FLAGS) > /dev/null)
+
```

Comment:
> Not sure, what it really does

Stack is for Haskell what Bundler is for Ruby. `stack setup` and `stack build --only-dependencies` together are like `bundle install`. The first command installs GHC and the standard libs (specified by the key `resolver` in the stack.yaml) to `~/.stack/` and the second command installs extra-dependencies (specific to the project; not included in the standard libs; specified by the key `extra-deps`) to `./.stack-work/`.

It's no problem to remove `./.stack-work` again with a `clean_stack` target.